### PR TITLE
[AlloyV2] Improve contact searching & creation; improve handling of large numbers of Alloy search results

### PR DESCRIPTION
**Pull request by davea**
Despite the query potentially having more than 2000 results, Alloy will raise an error if you try and fetch pages numbered beyond 100. In this case we recalculate the page size to stay below the 100 pages limit.

Also updates the contact search code to perform a narrower search on specific attributes.

### CODE REVIEW

File: perllib/Integrations/AlloyV2.pm [hunk 1]
```diff
@@ -242,7 +243,9 @@ sub search {
     my $result_count = $stats->{result};
     return [] unless $result_count;
 
-    my $pages = int( $result_count / 20 ) + 1;
+    my $pageSize = 20;
+    my $maxPages = 100;
+    my $pages = min($maxPages, int( $result_count / $pageSize ) + 1);
```
Comment (struan) [inline perllib/Integrations/AlloyV2.pm:251]: If the number of pages is > 100 we could divide the result_count by 100 and use that for the page size?

**Review by struan — APPROVED** [review]
I noticed this and was sure I'd fixed it but apparently not.

File: perllib/Integrations/AlloyV2.pm (hunk 1 above)
Comment (davea) [reply perllib/Integrations/AlloyV2.pm]: That's a better idea, yes! Done in 693b1d522849c9b623a1e0ecc59486ca1f29168c.

**Review by struan — APPROVED** [review]
👍

**Comment by davea** [comment]
Merged in ef5fac66271d6706eb8d9a2bc128f41c02a50bfd.

**Pull request closed without merging**